src/agentscope/service/web/wiki.py

```diff
@@ -0,0 +1,471 @@
+# -*- coding: utf-8 -*-
+"""
+Search contents from WikiPedia,
+including texts, categories, infotable, table,...
+"""
+
+import re
+
+# import requests
+from bs4 import BeautifulSoup
+
+from agentscope.service.service_response import (
+    ServiceResponse,
+    ServiceExecStatus,
+)
+from agentscope.utils.common import requests_get
+
+
+def wiki_api(params: dict) -> dict:
+    """Scratch information via Wiki API"""
+    url = "https://en.wikipedia.org/w/api.php"
+    return requests_get(url, params=params)
+
+
+def _check_entity_existence(entity: str) -> ServiceResponse:
+    search_params = {
+        "action": "query",
+        "list": "search",
+        "srsearch": entity,
+        "format": "json",
+    }
+
+    search_data = wiki_api(search_params)
+
+    if "query" in search_data and search_data["query"]["search"]:
```

Comment:
is it possible that "search" does not exist in the search_data["query"]?

Reply:
The structure of search_data is: {'query':{'searchinfo':xxx, 'search':xxx,...}} so there is always a 'search' in 'query'.